lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,781 @@
+            res = builder.create<btor::RotateLOp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_ror: 
+            res = builder.create<btor::RotateROp>(unknownLoc, 
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_saddo: 
+            res = builder.create<btor::SAddOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_sdivo: 
+            res = builder.create<btor::SDivOverflowOp>(unknownLoc,
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_smulo: 
+            res = builder.create<btor::SMulOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_ssubo: 
+            res = builder.create<btor::SSubOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_uaddo: 
+            res = builder.create<btor::UAddOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_umulo: 
+            res = builder.create<btor::UMulOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+        case BTOR2_TAG_usubo: 
+            res = builder.create<btor::USubOverflowOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]), 
+                                        cache.at(kids[1]));
+        break;
+
+        // unary ops
+        case BTOR2_TAG_const: {
+            auto opType = builder.getIntegerType( line->sort.bitvec.width );
+            res = builder.create<btor::ConstantOp>(unknownLoc, opType, 
+                                        builder.getIntegerAttr(
+                                            opType,
+                                            line->constant[0] - '0'));
```

Comment:
do not understand the logic of subtracting a character. Is that a fancy string to integer conversion?

Reply:
Will look at `llvm::APInt`